packages/website/src/components/lib/jsonSchema.ts

```diff
@@ -176,30 +176,33 @@ export function getTypescriptOptions(): DescribedOptionDeclaration[] {
  * Get the JSON schema for the typescript config
  */
 export function getTypescriptJsonSchema(): JSONSchema4 {
-  const properties = getTypescriptOptions().reduce((options, item) => {
-    if (item.type === 'boolean') {
-      options[item.name] = {
-        type: 'boolean',
-        description: item.description.message,
-      };
-    } else if (item.type === 'list' && item.element?.type instanceof Map) {
-      options[item.name] = {
-        type: 'array',
-        items: {
+  const properties = Object.fromEntries(
+    getTypescriptOptions().flatMap(item => {
+      let value;
+      if (item.type === 'boolean') {
+        value = {
+          type: 'boolean',
+          description: item.description.message,
+        };
+      } else if (item.type === 'list' && item.element?.type instanceof Map) {
+        value = {
+          type: 'array',
+          items: {
+            type: 'string',
+            enum: Array.from(item.element.type.keys()),
+          },
+          description: item.description.message,
+        };
+      } else if (item.type instanceof Map) {
+        value = {
           type: 'string',
-          enum: Array.from(item.element.type.keys()),
-        },
-        description: item.description.message,
-      };
-    } else if (item.type instanceof Map) {
-      options[item.name] = {
-        type: 'string',
-        description: item.description.message,
-        enum: Array.from(item.type.keys()),
-      };
-    }
-    return options;
-  }, {});
+          description: item.description.message,
+          enum: Array.from(item.type.keys()),
+        };
+      }
+      return value ? [[item.name, value]] : [];
```

Comment:
(optional) Similar to a previous comment, I'd much prefer a conditional push for these rather than flatmap with a 1- or 0-length array.

Reply:
Updated to use `.map.filter` because I like functional methods 😁